File: NetDeck/SmallBrowserCell.m
//
//  SmallBrowserCell.m
//  Net Deck
//
//  Created by Gereon Steffens on 08.08.14.
//  Copyright © 2016 Gereon Steffens. All rights reserved.
//

#import "SmallBrowserCell.h"

@implementation SmallBrowserCell

- (void)awakeFromNib
{
    [super awakeFromNib];
    
    self.pips = [SmallPipsView create];
    
    [self.pipsView addSubview:self.pips];
}

-(void) setCard:(Card *)card
{
    [super setCard:card];
    if (card.unique)
    {
        self.nameLabel.text = [NSString stringWithFormat:@"%@ •", card.name];
    }
    else
    {
        self.nameLabel.text = card.name;
    }
    
    NSInteger value = card.type == NRCardTypeAgenda ? card.agendaPoints : card.influence;
    [self.pips setWithValue:value color:card.factionColor];
    self.factionLabel.text = card.factionStr;
}

@end

File: NetDeck/SmallBrowserCell.swift
//
//  SmallBrowserCell.swift
//  NetDeck
//
//  Created by Gereon Steffens on 31.12.16.
//  Copyright © 2019 Gereon Steffens. All rights reserved.
//

import UIKit
import SwiftyUserDefaults

class SmallBrowserCell: BrowserCell {
    
    @IBOutlet weak var factionLabel: UILabel!
    @IBOutlet weak var pipsView: UIView!
    
    var pips: SmallPipsView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.pips = SmallPipsView.create()
        
        self.pipsView.layer.cornerRadius = 2
        self.pipsView.layer.masksToBounds = true
        
        self.pipsView.addSubview(self.pips)
    }
    
    override func prepareForReuse() {
        super.prepareForReuse()
        
        self.pipsView.backgroundColor = .white
    }
    
    override func setCard(_ card: Card) {
        let mwl = MWLManager.activeMWL
        self.nameLabel.text = card.displayName(mwl)
        
        let value = card.type == .agenda ? card.agendaPoints : card.influence
        self.pips.set(value: value, color: card.factionColor)
        self.factionLabel.text = card.factionStr
        
        if card.mwlPenalty(mwl) != 0 {
            self.pipsView.backgroundColor = UIColor(rgb: 0xf5f5f5)
        }
    }
}
